ui/src/features/project/pipelines/images.tsx

```diff
@@ -122,12 +122,20 @@ export const Images = ({ projectName, stages }: { projectName: string; stages: S
   }, [stages]);
 
   const [imageURL, setImageURL] = useState(images.keys().next().value as string);
-  const image = imageURL && images.get(imageURL);
+  const [curImage, setCurImage] = useState(imageURL && images.get(imageURL));
   const [showHistory, setShowHistory] = useLocalStorage(`${projectName}-show-history`, false);
 
+  useEffect(() => {
+    setImageURL(images.keys().next().value as string);
+  }, [images]);
+
+  useEffect(() => {
+    setCurImage(images.get(imageURL));
+  }, [imageURL]);
+
```

Comment:
I think you can use `useMemo` here.

Reply:
We can for `curImage` but not `imageURL` because we need to `setImageURL` elsewhere as well